### PR TITLE
feat: [CDS-86801]: support S3 manifest store in serverlessAwsLambda service defination

### PR DESCRIPTION
**Pull request by shivanand-harness**
### Summary
Harness Store would be supported for both the manifest types in Serverless V2 Swimlane
1. serverless lambda manifest
2. values YAML

scenarios tested
1. services
2. templates
3. pipelines (with and without runtime inputs)
4. input sets
5. triggers

YAML for 
1. serverless lambda manifest
```
manifest:
  identifier: id1
  type: ServerlessAwsLambda
  spec:
    store:
      type: S3
      spec:
        connectorRef: account.asws_Account
        region: us-east-1
        bucketName: my-first-serverless-proj-serverlessdeploymentbuck-pos1c8m0h4fh
        paths:
          - path.zip
    configOverridePath: test/path.yaml
```
2. values YAML
```
manifest:
  identifier: id2
  type: Values
  spec:
    store:
      type: S3
      spec:
        connectorRef: account.asws_Account
        region: us-east-1
        bucketName: do-not-delete-aws-cd-prod3
        paths:
          - path.zip
```

#### Screenshots
Services with fixed input

https://github.com/harness/harness-core-ui/assets/148759765/f0fb894a-fa22-44f8-b4ee-d66122b8af07

Services with runtime input

https://github.com/harness/harness-core-ui/assets/148759765/24f03d42-a152-47d2-ba56-9166614677ef

Pipeline with runtime inputs

https://github.com/harness/harness-core-ui/assets/148759765/e260ac2c-4efa-4284-abdc-b37ba266397f

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by shivanand-harness** [comment]
retrigger standards

File: src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx [hunk 1]
```diff
@@ -457,9 +457,14 @@ function ManifestListView({
...
         )
         break
-      case selectedManifest === ManifestDataType.ServerlessAwsLambda && manifestStore === ManifestStoreMap.S3:
+      case [ManifestDataType.ServerlessAwsLambda, ManifestDataType.Values].includes(
+        selectedManifest as ManifestTypes
+      ) && manifestStore === ManifestStoreMap.S3:
         manifestDetailStep = (
           <ServerlessLambdaWithS3
+            enableFields={{
+              configOverridePath: [ManifestDataType.ServerlessAwsLambda].includes(selectedManifest as ManifestTypes)
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx:466]: Is this additional `includes` check required? It only reaches here if the case at 460 with the `includes` check satisfies.

**Review by rameshx — DISMISSED** [review]

File: src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/ServerlessLambdaWithS3/ServerlessLambdaWithS3.tsx [hunk 2]
```diff
@@ -422,118 +426,124 @@ export function ServerlessLambdaWithS3({
...
-                      options={regions}
-                      style={{ alignSelf: 'center', marginBottom: 3 }}
-                      value={formik.values?.region as string}
-                      type="String"
-                      variableName="region"
-                      showRequiredField={false}
-                      showDefaultField={false}
-                      onChange={value => {
-                        formik.setFieldValue('region', value)
+                {get(enableFields, 'region', true) && (
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/ServerlessLambdaWithS3/ServerlessLambdaWithS3.tsx:429]: Do we need to show this field even if region is not in enabled field as passing default value as true.

Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/ServerlessLambdaWithS3/ServerlessLambdaWithS3.tsx:429]: Looks like `region`, `bucketName` are not passed from anywhere this component is used. Do we need this condition for these fields?

**Review by chetanvekariya-harness — COMMENTED** [review]
Please write jest test cases.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]
Please write test cases.

File: src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/ServerlessLambdaWithS3/ServerlessLambdaWithS3.tsx (hunk 2 above)
Comment (shivanand-harness) [reply src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/ServerlessLambdaWithS3/ServerlessLambdaWithS3.tsx:429]: enableFields prop is optional, if want to show/hide based on condition then we can use enableFields, else all fields will be shown

Comment (shivanand-harness) [reply src/modules/70-pipeline/components/ManifestSelection/ManifestWizardSteps/ServerlessLambdaWithS3/ServerlessLambdaWithS3.tsx:429]: enableFields prop is optional, if want to show/hide based on condition then we can use enableFields, else all fields will be shown

**Comment by ishantmehtaharness** [comment]
PR has conflicts

File: src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx [hunk 3]
```diff
@@ -457,9 +457,14 @@ function ManifestListView({
           />
         )
         break
-      case selectedManifest === ManifestDataType.ServerlessAwsLambda && manifestStore === ManifestStoreMap.S3:
+      case [ManifestDataType.ServerlessAwsLambda, ManifestDataType.Values].includes(
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx:460]: This change is not covered using test cases.

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/ValuesYamlManifestSource/ValuesYamlManifestSource.tsx [hunk 4]
```diff
@@ -18,6 +20,11 @@ export class ValuesYamlManifestSource extends ManifestSourceBase<ManifestSourceR
       return null
     }
 
+    const manifestStoreType = get(props.template, `${props.manifestPath}.spec.store.type`, null)
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/ValuesYamlManifestSource/ValuesYamlManifestSource.tsx:23]: This change is not covered in test case.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
looks good to me

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/ValuesYamlManifestSource/ValuesYamlManifestSource.tsx (hunk 4 above)
Comment (shivanand-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/ValuesYamlManifestSource/ValuesYamlManifestSource.tsx:23]: covered tests

File: src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx (hunk 3 above)
Comment (shivanand-harness) [reply src/modules/70-pipeline/components/ManifestSelection/ManifestListView/ManifestListView.tsx:460]: covered

**Review by vikrant-harness — APPROVED** [review]

**Review by namanharness — APPROVED** [review]

**Pull request merged**